doc/source/docs/howto_document.rst

```diff
@@ -73,3 +73,219 @@ Please use the numpydoc `formatting standard`_ as shown in their example_
 
 .. _`formatting standard`: https://numpydoc.readthedocs.io/en/latest/format.html
 .. _example: https://numpydoc.readthedocs.io/en/latest/example.html
+
+.. _doc_c_code:
+
+Documenting C/C++ Code
+**********************
+
+Recently, NumPy headed out to use Doxygen_ along with Sphinx due to the urgent need
+to generate C/C++ API reference documentation from comment blocks, especially with
+the most recent expansion in the use of SIMD, and also due to the future reliance on C++.
+Thanks to Breathe_, we were able to bind both worlds together at the lowest cost.
+
+**It takes three steps to complete the documentation process**:
+
+1. Writing the comment blocks
+-----------------------------
+
+Although there is still no commenting style set to follow, the Javadoc
```

Comment:
Can we have a link to the Javadoc standard?

Reply:
It wouldn't be very informative; since a lot of tags are not present in `doxygen`, like `@serialData`. Essentially in context it means do not use the alternative `doxygen` style format. So `\brief` is wrong, `@brief` is correct. The examples later clarify this usage.

Reply:
Can we add the following link as a reference at least?
https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html

"A lot" sounds exaggerate, I would rather say "a few" instead. Well as long you keep the asterisks on the left and using the Javadoc tags names then you're following the Javadoc style, I think using the Qtdoc tag starter `\` or stick to Javadoc tag starter `@` would make any difference. However, Javadoc doesn't require tags `@brief` and `@details` which allows documenting most of our current comment blocks without making any changes.

Reply:
Does that link make any sense to other contributors? Personally I skimmed it and it seems to reference API specifications which aren't relevant.. In fact most of the document seems spurious (Package level comments?)

We should feature the canonical:
https://www.doxygen.nl/manual/docblocks.html

More prominently.

Reply:
Javadoc is pretty known by most of the C++ developers, even libstdc++ using it. So just keeping https://www.doxygen.nl/manual/docblocks.html is enough for now.